--- before/python/battle_2_004_twistedv2.py
+++ after/python/battle_2_004_twistedv2.py
@@ -14,6 +14,10 @@
 """
 import numpy as np
 from core import Model
+from Simo1993_Tri6_ShellElement_v2 import Simo1993_Tri6_ShellElement_v2
+from MacNeal_MH6T_Tri_v3 import MacNeal_MH6T_Tri_v3
+from MITC6_Tri_v4 import MITC6_Tri_v4
+from Rezaiee2017_Tri6_v3 import Rezaiee2017_Tri6_ShellElement_v3
 from functools import partial
 
 # Q8 Python elements (Base classes)
@@ -34,13 +38,7 @@
 from  KikuchiMacNeal_MITC8_v1         import  KikuchiMacNeal_MITC8_v1
 
 # Dedicated T6 implementation (Base classes)
-from MITC6_Tri_v1 import MITC6_Tri_v1
-from MITC6_Tri_v0 import MITC6_Tri_v0
-from MacNeal_MH6T_Tri_v0 import MacNeal_MH6T_Tri_v0
-from MacNeal_MH6T_Tri_v1 import MacNeal_MH6T_Tri_v1
-
-from Rezaiee2017_Tri6_Final import Rezaiee2017_Tri6_ShellElement
-from Rezaiee2017_Tri6_v2 import Rezaiee2017_Tri6_ShellElement_v2
+
 
 # ---------------------------------------------------------------------
 # ADAPTER IMPORTS (Menggunakan elemen yang sudah diperbaiki)
@@ -76,13 +74,10 @@
 }
 
 T6_ELEMENTS = {
-    "SimoT6": SimoT6v18,           # <-- PAKAI ADAPTER (Fix get_node_normal)
-   "MITC6_v0" : MITC6_Tri_v0,
-    "MITC6_v1" : MITC6_Tri_v1,
-    "Rezaiee_v1" : Rezaiee2017_Tri6_ShellElement,
-    "Rezaiee_v2" : Rezaiee2017_Tri6_ShellElement_v2,
-    "MacNeal_MH6T_v0":MacNeal_MH6T_Tri_v0,
-    "MacNeal_MH6T": MacNeal_MH6T_Tri_v1,
+    "SIMOT6": Simo1993_Tri6_ShellElement_v2,
+    "MITC6": MITC6_Tri_v4,
+    "MH6T": MacNeal_MH6T_Tri_v3,
+    "REZAIEE": Rezaiee2017_Tri6_ShellElement_v3,
 }
 
 E = 29.0e6
