--- before/python/battle_2_006_scordelisv2.py
+++ after/python/battle_2_006_scordelisv2.py
@@ -28,6 +28,10 @@
 """
 import numpy as np
 from core import Model
+from Simo1993_Tri6_ShellElement_v2 import Simo1993_Tri6_ShellElement_v2
+from MacNeal_MH6T_Tri_v3 import MacNeal_MH6T_Tri_v3
+from MITC6_Tri_v4 import MITC6_Tri_v4
+from Rezaiee2017_Tri6_v3 import Rezaiee2017_Tri6_ShellElement_v3
 
 # Q8 Python elements (Base classes)
 from ANS8_BDG6_ShellElement import ANS8_BDG6_ShellElement
@@ -70,10 +74,10 @@
 }
 
 T6_ELEMENTS = {
-    "SimoT6": SimoT6v18,           # <-- PAKAI ADAPTER (Fix get_node_normal)
-    "MITC6_v1" : MITC6_Tri_v1,
-    "Rezaiee" : Rezaiee2017_Tri6_ShellElement,
-    "MacNeal_MH6T": MacNeal_MH6T_Tri_v1,
+    "SIMOT6": Simo1993_Tri6_ShellElement_v2,
+    "MITC6": MITC6_Tri_v4,
+    "MH6T": MacNeal_MH6T_Tri_v3,
+    "REZAIEE": Rezaiee2017_Tri6_ShellElement_v3,
 }
 
 # ==================================================================
